bson.cpp

```diff
@@ -150,6 +157,14 @@ void VariantToBsonConverter::convertElement(bson_t *bson, const char *key, Varia
 		case KindOfRef:
 			convertElement(bson, key, *v.getRefData());
 			break;
+#if HIPPO_HHVM_VERSION >= 31700
+		case KindOfPersistentVec:
+		case KindOfPersistentDict:
+		case KindOfPersistentKeyset:
+		case KindOfVec:
+		case KindOfDict:
+		case KindOfKeyset:
+#endif
```

Comment:
Exception below still says "Got unsupported type 'resource'".

Wouldn't it make more sense to move the failure case to a `default:` after all handled cases? That would obviate the need to list these types explicitly.